psi4/driver/procrouting/diis.py

```diff
@@ -0,0 +1,212 @@
+from enum import Enum
+from psi4 import core
+from psi4.driver import psifiles as psif
+
+import numpy as np
+
+class RemovalPolicy(Enum):
+    LargestError = 1
+    OldestAdded = 2
+
+class StoragePolicy(Enum):
+    InCore = 1
+    OnDisk = 2
+
+def axpy(y, alpha, x):
+    if isinstance(y, (core.Matrix, core.Vector)):
+        y.axpy(alpha, x)
+    elif isinstance(y, (core.dpdbuf4, core.dpdfile2)):
+        y.axpy_matrix(x, alpha)
```

Comment:
whoa, both `(alpha, x)` and `(x, alpha)`. typical development at different times :-)

Reply:
`libmints` wants `(alpha, x)` and `libdpd` wants `(x, alpha)`. But this particular function is one I created to pass `libmints` objects to `libdpd`. As the functions lived in `libdpd`, I opted to have them use `libdpd` convention, but I can go the other way.